src/models/Settings.js

```diff
@@ -238,6 +247,57 @@ function (Okta, Q, Errors, BrowserFeatures, Util, Logger, config) {
           return !!(socialAuthConfigured && idpDisplay.toUpperCase() === 'PRIMARY');
         },
         cache: true
+      },
+      configuredIdpsOptionsSet: {
+        deps: ['configuredSocialIdps'],
+        fn: function (configuredSocialIdps) {
+          _.each(configuredSocialIdps, function(idp) {
+            var type = idp.type.toLowerCase();
+            //set required properties
+            idp['attr'] = 'social-auth-' + type + '-button';
+            idp['className'] = 'social-auth-button ' + idp['attr'];
+            idp['title'] = Okta.loc('socialauth.' + type + '.label');
+            idp['click'] = socialAuthClickHandler;
+          })
+          return configuredSocialIdps;
+        },
+        cache: true
+      },
+
+      //custom buttons
+      customButtonsOptionsSet: {
+        deps: ['customButtons'],
+        fn: function (customButtons) {
+          _.each(customButtons, function(button) {
+            //add custom auth class name
+            button['className'] = button['className'] + ' default-custom-button';
+          });
+          return customButtons;
+        },
+        cache: true
+      },
+
+      configuredButtons: {
+        deps: ['customButtonsOptionsSet', 'configuredIdpsOptionsSet', 'socialAuthConfigured'],
+        fn: function(customButtonsOptionsSet, configuredIdpsOptionsSet, socialAuthConfigured) {
+          var buttonArray = [];
+          if(socialAuthConfigured) {
```

Comment:
Yeah, just reiterating - since the customButtonOptions and idpOptions are only really used for this (I'm guessing), let's just pack it all in one derived property.